### PR TITLE
Use [CORE]NEURON_BRANCH from PR descriptions in Azure CI

### PR DESCRIPTION
**Pull request by olupton**
- Update NEURON submodule commit of CoreNEURON using `CORENEURON_BRANCH`.
- Use `gcc-8` when we're using `g++-8`.

CI_BRANCHES:CORENEURON_BRANCH=olupton/canary

### CODE REVIEW

**Comment by olupton** [comment]
Note the name of the jobs has changed, because there is now a `build` stage, so we will have to update the required statuses in the GitHub repo configuration:
Screenshot 2022-05-09 at 09 44 47

File: azure-pipelines.yml [hunk 1]
```diff
@@ -21,261 +21,320 @@ trigger:
...
+        if pr_body is not None:
+          for pr_body_line in pr_body.splitlines():
+            if not pr_body_line.startswith('CI_BRANCHES:'): continue
+            for config_term in pr_body_line[12:].split(','):
+              pat.sub(parse_term, config_term)
+        END_SCRIPT
+        echo '----'
+        cat parse_description.py
+        echo '----'
+        python parse_description.py
```
Comment (olupton) [inline azure-pipelines.yml:60]: This is the main addition to the file.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -21,261 +21,320 @@ trigger:
...
+        export CC='gcc-8'
+        export CXX='g++-8'
+        git clone${NEURON_BRANCH:+ --branch }${NEURON_BRANCH} --single-branch https://github.com/neuronsimulator/nrn.git
+        if [[ -n "${CORENEURON_BRANCH}" ]]; then
+          pushd nrn
+          git submodule update --init external/coreneuron
+          cd external/coreneuron
+          git checkout ${CORENEURON_BRANCH}
+          popd
+        fi
```
Comment (olupton) [inline azure-pipelines.yml:126]: This has changed.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -21,261 +21,320 @@ trigger:
...
+          exit 1
+        fi
+        make install #this is needed for the integration tests
+        env CTEST_OUTPUT_ON_FAILURE=1 make test
+      env:
+        CMAKE_PKG: 'cmake-3.15.0-Linux-x86_64'
+      displayName: 'Build and Run Unit Tests'
+    - script: |
+        export PATH=$(pwd)/$CMAKE_PKG/bin:/home/vsts/.local/bin:$PATH
+        export CC='gcc-8'
```
Comment (olupton) [inline azure-pipelines.yml:117]: This was added.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -21,261 +21,320 @@ trigger:
...
+        fi
+        ./bin/nrnivmodl-core $(Build.Repository.LocalPath)/test/integration/mod
+      env:
+        CMAKE_PKG: 'cmake-3.15.0-Linux-x86_64'
+        SHELL: 'bash'
+      displayName: 'Build Neuron and Run Integration Tests'
+    - script: |
+        export PATH=$(pwd)/$CMAKE_PKG/bin:/home/vsts/.local/bin:$PATH
+        export CXX='g++-8'
+        git clone${CORENEURON_BRANCH:+ --branch }${CORENEURON_BRANCH} https://github.com/BlueBrain/CoreNeuron.git
```
Comment (olupton) [inline azure-pipelines.yml:150]: This was changed.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -21,261 +21,320 @@ trigger:
...
+        export CC='gcc-8'
+        export CXX='g++-8'
+        git clone${NEURON_BRANCH:+ --branch }${NEURON_BRANCH} --single-branch https://github.com/neuronsimulator/nrn.git
+        if [[ -n "${CORENEURON_BRANCH}" ]]; then
+          pushd nrn
+          git submodule update --init external/coreneuron
+          cd external/coreneuron
+          git checkout ${CORENEURON_BRANCH}
+          popd
+        fi
```
Comment (olupton) [inline azure-pipelines.yml:211]: This was changed.

File: azure-pipelines.yml [hunk 6]
```diff
@@ -21,261 +21,320 @@ trigger:
...
+        ctest --output-on-failure
+        if [ $? -ne 0 ]
+        then
+          exit 1
+        fi
+        make install
+      displayName: 'Build and Run Tests'
+    - script: |
+        export PATH=$(pwd)/$CMAKE_PKG/bin:/home/vsts/.local/bin:$PATH
+        export CC='gcc-8'
```
Comment (olupton) [inline azure-pipelines.yml:202]: Added.

File: azure-pipelines.yml [hunk 7]
```diff
@@ -21,261 +21,320 @@ trigger:
...
+        echo '----'
+        cat parse_description.py
+        echo '----'
+        python parse_description.py
+      name: prdesc
+- stage: build
+  dependsOn: fetch_config
+  variables:
+    NEURON_BRANCH: $[stageDependencies.fetch_config.github.outputs['prdesc.NEURON_BRANCH']]
+    CORENEURON_BRANCH: $[stageDependencies.fetch_config.github.outputs['prdesc.CORENEURON_BRANCH']]
```
Comment (olupton) [inline azure-pipelines.yml:66]: Added.

**Comment by olupton** [comment]
You can see it's working in:
- https://dev.azure.com/pramodskumbhar/nmodl/_build/results?buildId=4015&view=logs&j=2e248e5f-1493-5238-3090-df78b23fac0c&t=ef84038e-b234-5e5e-7a21-aa3ea1dea55c&l=69
- https://dev.azure.com/pramodskumbhar/nmodl/_build/results?buildId=4015&view=logs&j=c2725d84-3b5d-515a-752a-7712601f670e&t=7da1f2c3-f259-5abe-900e-b32bdcb1312f&l=68
- https://dev.azure.com/pramodskumbhar/nmodl/_build/results?buildId=4015&view=logs&j=c2725d84-3b5d-515a-752a-7712601f670e&t=8b50b967-fcce-5e33-78fe-73c2e7d49d38&l=23
from https://github.com/BlueBrain/CoreNeuron/blob/olupton/canary/CMakeLists.txt.

**Review by alkino — APPROVED** [review]

**Review by iomaganaris — APPROVED** [review]
LGTM :+1:

**Pull request merged**